### PR TITLE
Install Certbot with package

### PR DESCRIPTION
**Pull request by exploide**
This PR addresses #18 

It is now possible to choose whether Certbot should be installed from Git or from package management. The latter is the default now, but if you like it the other way around, I can change this.

As a side effect, I restructured the tasks by using includes.

Beside that, I mentioned that Fedora is also a compatible platform. I guess this holds true for most other distributions, so I removed the note that this role is for EL and Debian/Ubuntu only.

### CODE REVIEW

**Comment by exploide** [comment]
I guess the failed tests are due to the changed default behavior. I did nothing on the tests as of now because I wanted to wait for you giving a statement if installation from package by default is fine.

**Comment by exploide** [comment]
@geerlingguy I incorporated the changes you requested in #20 

Feel free to do a further review and let me know what you think and what's missing.

File: meta/main.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ dependencies: []
 
 galaxy_info:
   author: geerlingguy
-  description: "Certbot (for Let's Encrypt) for RHEL/CentOS and Debian/Ubuntu."
+  description: "Installs and configures Certbot (for Let's Encrypt)."
   company: "Midwestern Mac, LLC"
   license: "license (BSD, MIT)"
   min_ansible_version: 1.8
```
Comment (oxyc) [inline meta/main.yml]: The `package` module in `install-with-package.yml` requires ansible 2.0+

File: tasks/install-from-source.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+- name: Clone Certbot into configured directory.
+  git:
+    repo: "{{ certbot_repo }}"
+    dest: "{{ certbot_dir }}"
+    version: "{{ certbot_version }}"
+    update: "{{ certbot_keep_updated }}"
+    force: yes
```
Comment (oxyc) [inline tasks/install-from-source.yml]: I think most of @geerlingguy other install-from-source tasks either expose a `clone_depth` variable, or default it to 1 for faster clones. Maybe add it here too?

- https://github.com/geerlingguy/ansible-role-drush/blob/master/tasks/source-install.yml#L9
- https://github.com/geerlingguy/ansible-role-php-redis/blob/master/tasks/install-from-source.yml#L8
- https://github.com/geerlingguy/ansible-role-php/blob/master/tasks/install-from-source.yml#L85

File: meta/main.yml (hunk 1 above)
Comment (exploide) [inline meta/main.yml]: good catch! fixed it

File: tasks/install-from-source.yml (hunk 2 above)
Comment (exploide) [inline tasks/install-from-source.yml]: I just copied it over from the current implementation of this role. If such an option is wanted, one could introduce it but this seems to be out of scope for this PR.

File: tasks/install-with-package.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Install Certbot.
+  package: name=certbot state=present
```
Comment (oxyc) [inline tasks/install-with-package.yml]: The package is called `letsencrypt` on Ubuntu. And so is the binary for `certbot_script`.

Comment (oxyc) [inline tasks/install-with-package.yml]: Also to get the tests to pass you can change the travis setup to test the install from source scenario on centos6 (as there's no certbot package available anyways).

@geerlingguy recently made a quite [similar change in his `drush` role](https://github.com/geerlingguy/ansible-role-drush/commit/1fd265dc3c156a17233c8cb6dfcefdd4f43288c4). Have a look at the `.travis.yml` and `tests/test-source-install.yml` changes.

Comment (exploide) [inline tasks/install-with-package.yml]: For Debian it is certbot (with needed jessie-backports). Latest Ubuntu (16.10) also comes with certbot. Ubuntu 16.04 still calls this letsencrypt :(

Several options:
- ignore non-up-to-date distributions
- introduce a simple variable that allows to set the package name
- introduce more complexity by crafting distribution specific variable files that are included on demand
- do a quick and dirty `when` workaround for ubuntu and put letsencrypt in there

what do you think @geerlingguy ?

Comment (exploide) [inline tasks/install-with-package.yml]: The test setup slightly confuses me xD I would need a bit more time to adapt this. Feel free to fix it and open a PR against my branch. Otherwise geerlingguy could fix it.

Comment (oxyc) [inline tasks/install-with-package.yml]: ubuntu at it again.... :(

Comment (oxyc) [inline tasks/install-with-package.yml]: Mehh seems Ubuntu 1404 doesnt have the package either... This is one distro I think _might_ need it to work out of the box.

Comment (exploide) [inline tasks/install-with-package.yml]: 14.04 hasn't even letsencrpyt as it seems. In this case users simply need to set `certbot_install_from_source: yes` and rely on the Git installation. That's what it's good for :)

Comment (geerlingguy) [inline tasks/install-with-package.yml]: Yeah, I'm fine with defaulting to package install, but adding a note in the README about current LTS distros that will require install from source.

I just want to make sure we have the tests for each supported major LTS release (centos6, centos7, ubuntu 14/16, debian 8).

Comment (exploide) [inline tasks/install-with-package.yml]: Just had a look one these distributions and added the respective note to the README.

In the meantime, @oxyc commited changes to fix the test setup.

**Comment by peterjanes** [comment]
Looks like everything needs to test with `source-install` except for `centos7`.

File: tasks/install-with-package.yml (hunk 3 above)
Comment (peterjanes) [inline tasks/install-with-package.yml]: I'm guessing it would be better to update the failing tests to use from-source at this point? Or is the plan to continue updating the project to use the alternative repositories and/or alternative binary names for the failing distros?

Comment (oxyc) [inline tasks/install-with-package.yml]: 12.04 could probably use from-source as well. But in my opinion, at least Ubuntu 16.04 should work out of the box, no? @geerlingguy

Comment (oxyc) [inline tasks/install-with-package.yml]: @geerlingguy I opened up a PR against @exploide branch that would add support for Ubuntu 16.04 out of the box. What do you think? https://github.com/exploide/ansible-role-certbot/pull/3

Comment (exploide) [inline tasks/install-with-package.yml]: I would merge this if @geerlingguy is fine with it. Not sure if this is the way to go but definitely want this PR to have a happy end. 😄

Comment (oxyc) [inline tasks/install-with-package.yml]: btw the reason I'm pushing for out of the box support for 16.04 is that there's a downstream [issue](https://github.com/geerlingguy/drupal-vm/issues/1251) in Drupal VM to add this role.

Comment (geerlingguy) [inline tasks/install-with-package.yml]: @oxyc - I'd do it, yes. We can default any of the latest OSes with a good package setup to use the package install instead of the source install.

Comment (exploide) [inline tasks/install-with-package.yml]: Merged the PR against my branch. Anything else missing?

**Comment by oxyc** [comment]
And now Debian. Maybe we should actually split stuff into:

`vars/Debian.yml`/`vars/RedHat.yml` or `init-RedHat.yml`/`init-Debian.yml`.

That might be useful for future features. Eg. https://github.com/geerlingguy/ansible-role-certbot/issues/12#issuecomment-289226914

> We should probably define a certbot_certificate_dir that points to the live/ subdir on all different distros. This way you just pass that on the the nginx/ansible role

File: tasks/install-from-source.yml (hunk 2 above)
Comment (geerlingguy) [inline tasks/install-from-source.yml]: Yeah, that could be done separately/later.

**Comment by geerlingguy** [comment]
Definitely would like to use the Debian/RedHat split config like I do with other roles.

**Comment by exploide** [comment]
@oxyc Are you working on it or should I?

**Comment by exploide** [comment]
Travis is happy now. I introduced distribution specific include_vars as suggested. Beside that I turned tests for Debian 8 to source installation because certbot and letsencrypt packages are only available from jessi-backports repository.

Further comments or are we done here? 😅

**Comment by geerlingguy** [comment]
Looks good, I'm going to tweak it a little I think, though; a couple language corrections in the documentation mostly. Also, this will require a major version bump, since it switches from git to package install (that's a breaking change).
